src/fmeval/eval_algorithms/context_quality.py

```diff
@@ -0,0 +1,291 @@
+from typing import Optional, List, Dict, Any
+import logging
+from dataclasses import dataclass
+
+from fmeval.constants import (
+    DatasetColumns,
+    MEAN,
+)
+from fmeval.data_loaders.data_config import DataConfig
+from fmeval.data_loaders.util import get_dataset
+from fmeval.eval_algorithms import EvalAlgorithm, EvalScore, EvalOutput
+from fmeval.eval_algorithms.common import evaluate_dataset
+from fmeval.eval_algorithms.eval_algorithm import (
+    EvalAlgorithmInterface,
+    EvalAlgorithmConfig,
+)
+from fmeval.eval_algorithms.save_strategy import SaveStrategy
+from fmeval.eval_algorithms.util import get_dataset_configs, validate_dataset
+from fmeval.model_runners.bedrock_model_runner import BedrockModelRunner
+from fmeval.model_runners.model_runner import ModelRunner
+from fmeval.transforms.transform import Transform
+from fmeval.transforms.util import validate_call
+from fmeval.model_runners.composers.composers import PromptComposer
+from fmeval.transforms.transform_pipeline import TransformPipeline
+from fmeval.util import get_eval_results_path
+
+logger = logging.getLogger(__name__)
+
+CONTEXT_PRECISION_SCORE = "context_precision_score"
+
+DEFAULT_CONTEXT_PRECISION_PROMPT_TEMPLATE = (
+    "Given question, answer and context verify if the context was useful in "
+    "arriving at the given answer. Give verdict as 1 if useful and 0 if "
+    "not. question: $model_input, answer: $target_output, "
+    "context: $retrieved_context."
+    "The verdict should only contain an integer, either 1 or 0, do not give an explanation."
```

Comment:
Any reason why the model is not supposed to give an explanation for it's verdict? In practice, forcing the model to generate reasoning steps generally leads to a more accurate answer (for example see chain-of-thought prompting).

Reply:
This was considering we only parse the verdict of "1" or "0" from the model response, so we don't need the explanation. I can update this prompt to ask the model to generate an explanation if it helps with accuracy.

Reply:
If we rely on the user to provide their own prompt this shouldn't be an issue. However, if we provide a default judge model and a default prompt, some experimental verification might be needed if forcing an explanation leads to better verdicts at the cost of increased output length in this case.